### ISSUE TITLE
NUnit2021 false positive for IEquatable<T>

### ISSUE THREAD

**Issue by pantosha** [nunit/nunit.analyzers#322]
NUnit2021 fires when different types `T1` and `T2` are compared, but `T1` implements `IEquatable<T2>`:

```csharp
using Microsoft.Extensions.Primitives;
using NUnit.Framework;

namespace UnitTests
{
    public class Tests
    {
        [Test]
        public void Test1()
        {
            var stringValues = new StringValues("abc");
            Assert.That(stringValues, Is.EqualTo("abc")); // <-- NUnit2021 fires here. Test passes ✅.
        }
    }
}
```

**Comment by Dreamescaper** [comment]
I'll take a look

**Comment by manfred-brands** [comment]
@Dreamescaper This is similar to the code for `IComparable<T>`. See ComparableTypesAnalyzer.IsIComparable.
